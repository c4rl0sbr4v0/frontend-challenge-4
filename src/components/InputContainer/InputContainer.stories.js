import React from 'react';
import InputContainer from './index';

export default {
  title: 'Components/InputContainer',
  component: InputContainer,
};

export const Default = () => <InputContainer>Container</InputContainer>;
