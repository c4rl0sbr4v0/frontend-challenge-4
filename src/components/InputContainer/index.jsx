import React from 'react';
import PropTypes from 'prop-types';

/**
 *  InputContainer: Render a Input Container
 */
const InputContainer = ({ children }) => {
  return (
    <div className="landbot-input-container">
      <div className="field">
        <div className="control">{children}</div>
      </div>
    </div>
  );
};

export default InputContainer;

InputContainer.propTypes = {
  children: PropTypes.node,
};
InputContainer.defaultProps = {
  children: '',
};
