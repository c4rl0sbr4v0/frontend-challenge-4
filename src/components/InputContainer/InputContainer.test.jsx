import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import InputContainer from '.';

test('Input Container Component', async () => {
  const { container } = render(
    <InputContainer>
      <div>ey</div>
    </InputContainer>,
  );
  expect(container.firstChild.classList.contains('landbot-input-container')).toBe(true);
  expect(container).toMatchSnapshot();
});
