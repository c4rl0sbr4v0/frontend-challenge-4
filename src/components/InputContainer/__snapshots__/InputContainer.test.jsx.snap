// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Container Component 1`] = `
<div>
  <div
    class="landbot-input-container"
  >
    <div
      class="field"
    >
      <div
        class="control"
      >
        <div>
          ey
        </div>
      </div>
    </div>
  </div>
</div>
`;
